File: api-tester/src/main/java/com/revenuecat/apitester/java/RecurrenceModeAPI.java
package com.revenuecat.apitester.java;

import com.revenuecat.purchases.models.RecurrenceMode;

@SuppressWarnings({"unused", "SpellCheckingInspection"})
final class RecurrenceModeAPI {
    static void check(final RecurrenceMode recurrenceMode) {
        switch (recurrenceMode) {
            case NON_RECURRING:
            case FINITE_RECURRING:
            case INFINITE_RECURRING:
            case UNKNOWN:
        }
    }
}

File: api-tester/src/main/java/com/revenuecat/apitester/kotlin/RecurrenceModeAPI.kt
package com.revenuecat.apitester.kotlin

import com.revenuecat.purchases.models.RecurrenceMode

@Suppress("unused", "UNUSED_VARIABLE")
private class RecurrenceModeAPI {
    fun check(recurrenceMode: RecurrenceMode) {
        when (recurrenceMode) {
            RecurrenceMode.INFINITE_RECURRING,
            RecurrenceMode.FINITE_RECURRING,
            RecurrenceMode.NON_RECURRING,
            RecurrenceMode.UNKNOWN,
            -> {
            }
        }.exhaustive
    }
}
